hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -330,9 +330,19 @@ case class HoodieFileIndex(spark: SparkSession,
             // fall back to listing all partitions
             case _: HoodieException => (false, listMatchingPartitionPaths(Seq.empty))
           }
+        } else if (isExpressionIndexEnabled) {
+          val expressionIndexSupport = new ExpressionIndexSupport(spark, schema, metadataConfig, metaClient)
```

Comment:
is there a way to tell `HoodieFileIndex` that EI was used in partition pruning, and if so, then avoid instantiating `ExpressionIndexSupport` twice?

Reply:
Addressed